_posts/2017-09-18-rnn-bucket-mxnet-R.md

```diff
@@ -0,0 +1,206 @@
+---
+layout: post
+title: "RNN made easy with MXNet R"
+date:   2017-09-18
+author: Jeremie Desgagne-Bouchard
+categories: rstats
+comments: true
+---
+
+This tutorial presents an example of application of RNN to text classification using padded and bucketed data to efficiently handle sequences of varying lengths. Some functionalities require running on a GPU with CUDA.
+
+Example based on sentiment analysis on the [IMDB data](http://ai.stanford.edu/~amaas/data/sentiment/).
+
+What's special about sequence modeling?
+---------------------------------------
+
+Whether working with times series or text at the character or word level, modeling sequences typically involves dealing with samples of varying length.
+
+To efficiently feed the Recurrent Neural Network (RNN) with samples of even length within each batch, two tricks can be used:
+
+-   Padding: fill the modeled sequences with an arbitrary word/character up to the longest sequence. This results in sequences of even lengths, but potentially of excessive size for an efficient training.
+
+![](https://raw.githubusercontent.com/dmlc/web-data/master/mxnet/blog_mxnet_R_rnn_bucket/pad-1.png)![](https://raw.githubusercontent.com/dmlc/web-data/master/mxnet/blog_mxnet_R_rnn_bucket/pad-2.png)
```

Comment:
Could you please include the sentence completely in the box? For this one and several plots below.